### PR TITLE
statesync for subnet-evm

### PR DESCRIPTION
**Pull request by darioush**
ports state-sync from coreth

Closes: https://github.com/ava-labs/subnet-evm/issues/158
e2e tests: https://github.com/ava-labs/subnet-evm/pull/190

### CODE REVIEW

File: plugin/evm/syncervm_test.go [hunk 1]
```diff
@@ -0,0 +1,511 @@
...
+	appSender.SendAppRequestF = func(nodeSet ids.NodeIDSet, requestID uint32, request []byte) error {
+		nodeID, hasItem := nodeSet.Pop()
+		if !hasItem {
+			t.Fatal("expected nodeSet to contain at least 1 nodeID")
+		}
+		go vmSetup.serverVM.AppRequest(nodeID, requestID, time.Now().Add(1*time.Second), request)
+		return nil
+	}
+	// Disable metrics to prevent duplicate registerer
+	configJSON := "{\"metrics-enabled\":false}"
```
Comment (darioush) [inline plugin/evm/syncervm_test.go:124]: should remove this, I think it no longer does anything

**Review by ceyonur — APPROVED** [review]
Some nits, other than that LGTM!
I also reapplied coreth changes in #216 on top of this PR. Seen no diff with coreth for state sync. Good work!

We should ship this and #126 together in the same release.

File: core/rawdb/accessors_state_sync.go [hunk 2]
```diff
@@ -0,0 +1,156 @@
+// (c) 2022, Ava Labs, Inc. All rights reserved.
+// See the file LICENSE for licensing terms.
+
+package rawdb
+
+import (
+	"github.com/ava-labs/subnet-evm/ethdb"
+	"github.com/ethereum/go-ethereum/common"
+	"github.com/ethereum/go-ethereum/core/rawdb"
```
Comment (aaronbuchwald) [inline core/rawdb/accessors_state_sync.go:9]:
```suggestion
	"github.com/ava-labs/coreth/core/rawdb"
```

Comment (darioush) [reply core/rawdb/accessors_state_sync.go:9]: good find

**Review by aaronbuchwald — APPROVED** [review]
LGTM - great work!

**Pull request merged**